src/collatz.rs

```diff
@@ -0,0 +1,82 @@
+use super::*;
+use log::debug;
+
+// EXAMPLE BUILDER
+// ================================================================================================
+
+pub fn get_example(start_value: usize) -> Example {
+    // convert starting value of the sequence into a field element
+    let start_value = Felt::new(start_value as u64);
+
+    // determine the expected result
+    let expected_result = compute_collatz_steps(start_value).as_int();
+
+    // construct the program which executes an unbounded loop to compute a Collatz sequence
+    // which starts with the provided value; the output of the program is the number of steps
+    // needed to reach the end of the sequence
+    let assembler = Assembler::new(true);
+    let program = assembler.compile(
+        "
+    begin
+        push.0 adv_push.1 dup push.1 neq
+        while.true
+            # counter
+            swap push.1 add swap 
+            
+            # actual check
+            dup push.2 u32checked_mod
+            if.true
```

Comment:
@bobbinth I cannot push this PR forward, because also here I have the problem described in https://github.com/maticnetwork/miden/issues/466

If I remove the `if.true` procedure, it works like a charm. If I keep the procedure I get 

```
thread 'thread 'collatz::test_collatz_example_failcollatz::test_collatz_example' panicked at '' panicked at 'main transition constraint 1 did not evaluate to ZERO at step 28main transition constraint 1 did not evaluate to ZERO at step 28'
```

At lease `push.2 u32checked_mod` must always be either `0` or `1` so the `if.true` should not fail